nucliadb_models/nucliadb_models/query.py

```diff
@@ -0,0 +1,427 @@
+                        ]
+                    },
+                    "minItems": 2,
+                },
+                "not": {
+                    "title": "Not",
+                    "anyOf": [{"type": "string"}, {"$ref": "#/definitions/Expression"}],
+                },
+            },
+        }
+    },
+}
+
+
+## Helper functions that make it easier to create expressions
+
+Value = str
+Expression = Dict[str, Any]
+Term = Union[Value, Expression]
+Terms = List[Term]
+TermsV1 = List[Value]
+
+
+def AND(*terms) -> Expression:
+    """
+    >>> AND("foo", "bar")
+    {'and': ['foo', 'bar']}
+    >>> AND(["foo", "bar"])
+    {'and': ['foo', 'bar']}
+    """
+    if isinstance(terms, tuple) and len(terms) == 1:
+        terms = terms[0]
+    elif not isinstance(terms, list):
+        terms = list(terms)
+    return {"and": terms}
+
+
+def OR(*terms) -> Expression:
+    """
+    >>> OR("foo", "bar")
+    {'or': ['foo', 'bar']}
+    >>> OR(["foo", "bar"])
+    {'or': ['foo', 'bar']}
+    """
+    if isinstance(terms, tuple) and len(terms) == 1:
+        terms = terms[0]
+    elif not isinstance(terms, list):
+        terms = list(terms)
+    return {"or": terms}
+
+
+def NOT(term: Term) -> Expression:
+    return {"not": term}
+
+
+# For those who prefer to use the Lucene syntax
+Must = AND
+Should = OR
+
+
+def MustNot(*terms):
+    return NOT(Must(*terms))
+
+
+def convert_to_v2(terms_v1: TermsV1) -> Expression:
+    """
+    Convert v1 filters to v2 filters
+    """
+    return AND(terms_v1)
+
+
+def andify(expression: Expression) -> Expression:
+    """
+    Convert an expression to a canonical form where all the 'or' operators
+    are replaced by 'and' operators with 'not' operators inside.
+
+    >>> andify({"or": ["/entity/GPE/Sevilla", "/entity/GPE/Madrid"]})
+    {'and': [{'not': '/entity/GPE/Sevilla'}, {'not': '/entity/GPE/Madrid'}]}
+
+    >>> andify({"or": ["/entity/GPE/Sevilla", {"not": "/entity/GPE/Madrid"}]})
+    {'and': [{'not': '/entity/GPE/Sevilla'}, {'not': {'not': '/entity/GPE/Madrid'}}]}
+    """
+    if isinstance(expression, str):
+        return expression
+    if "or" in expression:
+        return {"and": [{"not": andify(subexpr)} for subexpr in expression["or"]]}
+    elif "and" in expression:
+        return {"and": [andify(subexpr) for subexpr in expression["and"]]}
+    elif "not" in expression:
+        return {"not": andify(expression["not"])}
+    else:
+        return expression
+
+
+#######################
+
+
+class SearchRequest(BaseModel):
+    query: str
+    # Supports either v1 (list of terms) and v2 (expression)
+    filters: Optional[Union[TermsV1, Expression]] = Field(
+        default=None,
+        title="Filters",
+        description="Filters to apply to the query. It can be either a list of terms or an expression. Expressions are more powerful and allow to combine terms with 'and', 'or' and 'not' operators. See the documentation for more details.",
```

Comment:
```suggestion
        description="Filters to apply to the query. It can be either a list of terms or an expression. A list of terms will be considered as an 'and' filter for each term. Expressions are more powerful and allow to combine terms with 'and', 'or' and 'not' operators. See the documentation for more details.",
```